common/lsp/dummy-ls.cc

```diff
@@ -61,23 +61,25 @@ InitializeResult InitializeServer(const nlohmann::json &params) {
 
 int main(int argc, char *argv[]) {
 #ifdef _WIN32
+  // Windows messes with newlines by default. Fix this here.
   _setmode(_fileno(stdin), _O_BINARY);
+  _setmode(_fileno(stdout), _O_BINARY);
 #endif
 
   std::cerr << "Note: this dummy-ls is for testing." << std::endl;
 
   // Input and output is stdin and stdout
-  static constexpr int in_fd = 0;  // STDIN_FILENO
+  constexpr int kInputFD = 0;  // STDIN_FILENO, but Win does not have that macro
```

Comment:
would an `#ifndef STDIN_FILENO ...` help here?

Reply:
but then we have 3 lines of code that clutter up things.